cpu/samd21/periph/uart.c

```diff
@@ -100,41 +122,84 @@ int uart_init_blocking(uart_t uart, uint32_t baudrate)
             GCLK->CLKCTRL.reg = (uint16_t)((GCLK_CLKCTRL_CLKEN | GCLK_CLKCTRL_GEN_GCLK0 | (SERCOM0_GCLK_ID_SLOW << GCLK_CLKCTRL_ID_Pos)));
             while (GCLK->STATUS.bit.SYNCBUSY);
 
-            /* configure PINS to input/output*/
-            UART_0_PORT.DIRSET.reg = (1 << UART_0_TX_PIN);  /* tx's direction is output */
-            UART_0_PORT.PINCFG[UART_0_RX_PIN % 32].bit.INEN = true; /* buffer rx pin's value */
-
-            /* enable PMUX for pins and set to config D. See spec p. 12 */
-            UART_0_PORT.WRCONFIG.reg = PORT_WRCONFIG_WRPINCFG \
-                                        | PORT_WRCONFIG_WRPMUX \
-                                        | PORT_WRCONFIG_PMUX(0x3) \
-                                        | PORT_WRCONFIG_PMUXEN \
-                                        | UART_0_PINS;
-
-            UART_0_DEV.CTRLA.bit.ENABLE = 0; //Disable to write, need to sync tho
-            while(UART_0_DEV.SYNCBUSY.bit.ENABLE);
-
-            /* set to LSB, asynchronous mode without parity, PAD0 Tx, PAD1 Rx,
-             * 16x over-sampling, internal clk */
-            UART_0_DEV.CTRLA.reg = SERCOM_USART_CTRLA_DORD \
-                                    | SERCOM_USART_CTRLA_FORM(0x0) \
-                                    | SERCOM_USART_CTRLA_SAMPA(0x0) \
-                                    | SERCOM_USART_CTRLA_TXPO(0x0) \
-                                    | SERCOM_USART_CTRLA_RXPO(0x1) \
-                                    | SERCOM_USART_CTRLA_SAMPR(0x0) \
-                                    | SERCOM_USART_CTRLA_MODE_USART_INT_CLK;
-
-            /* Set baud rate */
-            UART_0_DEV.BAUD.bit.BAUD = baud_calculated;
-
-            /* enable receiver and transmitter, one stop bit*/
-            UART_0_DEV.CTRLB.reg = (SERCOM_USART_CTRLB_RXEN | SERCOM_USART_CTRLB_TXEN);
-            while(UART_0_DEV.SYNCBUSY.bit.CTRLB);
+            break;
+#endif
+#if UART_1_EN
+        case UART_1:
+            port_group = &UART_1_PORT;
+            uart_dev = &UART_1_DEV;
+            uart_ref_f = UART_1_REF_F;
+            tx_pin = UART_1_TX_PIN;
+            rx_pin = UART_1_RX_PIN;
+            pins = UART_1_PINS;
+
+            /* Turn on power manager for sercom5 */
+            PM->APBCMASK.reg |= PM_APBCMASK_SERCOM5;
+
+            /* Enable generic clock generator1 */
+            GCLK->GENCTRL.reg = (GCLK_GENCTRL_GENEN | (GCLK_CLKCTRL_ID(GCLK_CLKCTRL_GEN_GCLK1_Val)));
```

Comment:
I don't think this will work without specifying a source for the clock generator.